Framework/Algorithms/src/GetEiMonDet2.cpp

```diff
@@ -425,7 +425,11 @@ void GetEiMonDet2::parseIndices(std::vector<size_t> &detectorIndices,
         spectraListParser.parse(getProperty(PropertyNames::DETECTORS)));
     auto back = std::back_inserter(detectorIndices);
     std::copy(detectors.begin(), detectors.end(), back);
-    monitorIndex = getProperty(PropertyNames::MONITOR);
+    int monitor = getProperty(PropertyNames::MONITOR);
+    if (monitor < 0) {
```

Comment:
Sorry, I didn't realise (by looking only at the code changes in the PR) that the check was being applied to the case of a workspace index only. What I also didn't see was code duplication: properties DETECTORS and MONITOR get read 3 times.

To be honest I don't know whether a det id can be negative or not. May I ask you to refactor the method and get the properties only once before the `if` and process them afterwards. If you are also unsure about the sign of the IDs please revert your last change and do the check for workspace indices only.

What I also noticed is that the DETECTORS property is a string. Do you know if there is a reason for that? I think `ArrayProperty<int>` suites better to it.